src/searchengine/searchengine.cpp

```diff
@@ -511,29 +528,34 @@ void SearchEngine::appendSearchResult(const QString &line) {
     goToDescBtn->setEnabled(true);
 }
 
-void SearchEngine::closeTab(int index) {
+void SearchEngine::closeTab(int index)
+{
     // Search is run for active tab so if user decided to close it, then stop search
-    if (activeSearchTab && index == tabWidget->indexOf(activeSearchTab)) {
+    if (!currentSearchTab.isNull() && index == tabWidget->indexOf(currentSearchTab)) {
```

Comment:
Why it is current now? Search is connected to ```activeSearchTab```
If active is closed then search should be stopped. If it is not active, then there is no need to care of it.